homeassistant/components/media_player/_reproduce_state.py

```diff
@@ -0,0 +1,45 @@
+"""Module that groups code required to handle state restore for component."""
+from typing import Iterable
+
+from homeassistant.const import (
+    ATTR_STATE, SERVICE_MEDIA_PAUSE, SERVICE_MEDIA_PLAY, SERVICE_MEDIA_SEEK,
+    SERVICE_MEDIA_STOP, SERVICE_TURN_OFF, SERVICE_TURN_ON, SERVICE_VOLUME_MUTE,
+    SERVICE_VOLUME_SET, STATE_IDLE, STATE_OFF, STATE_ON, STATE_PAUSED,
+    STATE_PLAYING)
+from homeassistant.core import State
+from homeassistant.helpers.state import (
+    async_construct_service_schemas, async_reproduce_domain_state)
+from homeassistant.helpers.typing import HomeAssistantType
+from homeassistant.loader import bind_hass
+
+
+@bind_hass
+async def async_reproduce_states(hass: HomeAssistantType,
```

Comment:
We need to pass context so that state changes are attributes to the right initiator.

Reply:
Context is part of the State object already.

Reply:
context of the state object is the stored state, we need to pass the context of the one that invoked the action (either being a service call or automation/script)

Reply:
Btw. while i'm looking here. Do i need to support the blocking=false case?

Reply:
nah, we should drop that.